src/main/java/io/minio/MinioClient.java

```diff
@@ -172,7 +172,7 @@ public MinioClient(String endpoint) throws InvalidEndpointException, InvalidPort
    * @see #MinioClient(String endpoint, int port, String accessKey, String secretKey, boolean secure)
    */
   public MinioClient(URL url) throws NullPointerException, InvalidEndpointException, InvalidPortException {
-    this(url.toString(), 0, null, null, true);
+    this(url.toString(), 0, null, null, url.getProtocol().equals(Scheme.HTTPS.toString()));
```

Comment:
Its better to use like `Scheme.fromString(url.getProtocol()) == Scheme.HTTPS` because `Scheme.fromString()` works case insensitively.